packages/oscal-react-library/src/components/OSCALProperties.tsx

```diff
@@ -30,21 +30,40 @@ const OSCALPropertiesButton = styled(Button)(
 );
 
 /**
- * Accumulates all instances of third-party namespaces into a list.
+ * Sorts the namespaces by NIST namespaces first, orders third-party namespaces alphabetically,
+ * then orders by name.
  *
  * @param {Property[]} properties A list of properties
- * @returns {string[]} A list of third-party namespaces
+ * @returns {Property[] | undefined} A list of third-party namespaces
  */
-function getThirdPartyNamespaces(properties?: Property[]) {
-  const items: string[] = [""];
+function sortProperties(properties?: Property[]): Property[] | undefined {
+  const namespaceSortedProperties = properties?.sort((propA, propB) =>
+    namespaceOf(propA.ns) > namespaceOf(propB.ns) ? 1 : -1
+  );
+  const nistNamespaceFirstProperties = namespaceSortedProperties?.sort((propA, propB) =>
+    isNistNamespace(propA.ns) && !isNistNamespace(propB.ns) ? 1 : -1
+  );
+  const nameSortedProperties = nistNamespaceFirstProperties?.sort((propA, propB) =>
+    namespaceOf(propA.ns) === namespaceOf(propB.ns) && propA?.name > propB?.name ? 1 : -1
+  );
+
+  return nameSortedProperties;
+}
+
+/**
+ * Provides a complete list of unique namespaces for properties.
+ *
+ * @param {Property[]} properties A list of properties
+ * @returns {string[]} A list of namespaces
+ */
+function getNamespaces(properties?: Property[]): string[] {
+  const items: string[] = [];
 
-  properties
-    ?.filter((property: any) => !isNistNamespace(property?.ns))
-    .map((property: any) => {
-      if (!items.includes(property?.ns)) {
-        items.push(property?.ns);
-      }
-    });
+  properties?.map((property: any) => {
+    if (!items.includes(namespaceOf(property?.ns))) {
+      items.push(namespaceOf(property?.ns));
+    }
+  });
```

Comment:
If you're going to do this over the suggestion I had made, then this needs to be `forEach` not a `map`. You're `map`ing to void/undefined. Which is very ungood.